data-pipeline/tests/snapshots/compare_v05_trace_snapshot_test.json

```diff
@@ -0,0 +1,66 @@
+[
+  [
+    {
+      "service": "test-service",
+      "name": "test_name",
+      "resource": "test-resource",
+      "trace_id": 1234,
+      "span_id": 12342,
+      "parent_id": 12341,
+      "start": 1,
+      "duration": 5,
+      "error": 0,
+      "meta": {
+        "env": "test-env",
+        "service": "test-service",
+        "runtime-id": "test-runtime-id-value"
+      },
+      "metrics": {
+        "_dd_metric1": 1.1,
+        "_dd_metric2": 2.2
+      },
+      "type": ""
+    },
+    {
+      "service": "test-service",
+      "name": "test_name",
+      "resource": "test-resource",
+      "trace_id": 1234,
+      "span_id": 12343,
+      "parent_id": 12341,
+      "start": 1,
+      "duration": 5,
+      "error": 0,
+      "meta": {
+        "env": "test-env",
+        "service": "test-service",
+        "runtime-id": "test-runtime-id-value"
+      },
+      "metrics": {},
```

Comment:
Do the empty fields for `error`, `metrics` and `type` have to be present for v0.5? They do not for v0.4 and the test-agent doesn't complain about it if you include optional fields. There was a discussion around this [here](https://github.com/DataDog/libdatadog/pull/897#discussion_r1971934009).

Reply:
Yes, the [spec](https://github.com/DataDog/datadog-agent/blob/579d60cbfc8635559338b19fa855bf59b6d10f00/pkg/trace/api/version.go#L71) states that all elements should be there